elasticdl/pkg/common/embedding_table.go

```diff
@@ -14,24 +14,31 @@ func NewEmbeddingTable(name string, dim int64, initializer string) *EmbeddingTab
 	table.Name = name
 	table.Dim = dim
 	table.Initializer = initializer
-	table.EmbeddingVector = make(map[int64][]float32)
+	table.EmbeddingVector = make(map[int64]*Tensor)
 	return &table
 }
 
 // GetEmbeddingVector returns embedding vector giving an index
-func (table *EmbeddingTable) GetEmbeddingVector(index int64) []float32 {
+func (table *EmbeddingTable) GetEmbeddingVector(index int64) *Tensor {
 	if value, ok := table.EmbeddingVector[index]; ok {
 		return value
 	}
 	// TODO(qijun) only support zero initializer now
-	newVector := make([]float32, table.Dim)
+	newVector := NewVector(table.Dim)
 	table.EmbeddingVector[index] = newVector
 	return newVector
 }
 
+// SetEmbeddingVector returns embedding vector giving an index
+func (table *EmbeddingTable) SetEmbeddingVector(index int64, t *Tensor) bool {
+	_, ok := table.EmbeddingVector[index]
+	table.EmbeddingVector[index] = t
+	return ok
```

Comment:
Not very understand the logic here, why not just

```
table.EmbeddingVector[index] = t
```

Reply:
This function can tell if the index is previously occupied.

Reply:
removed